src/vulkan/vk-surface.cpp

```
@@ -131,15 +131,21 @@ Result SurfaceImpl::init(DeviceImpl* device, WindowHandle windowHandle)
-    {
+    // Derive the supported usage from the surface capabilities.
+    VkSurfaceCapabilitiesKHR surfaceCaps = {};
+    SLANG_VK_RETURN_ON_FAIL_REPORT(
+        api.vkGetPhysicalDeviceSurfaceCapabilitiesKHR(api.m_physicalDevice, m_surface, &surfaceCaps),
+        m_device
+    );
+    m_info.supportedUsage = TextureUsage::Present | TextureUsage::RenderTarget;
+    if (surfaceCaps.supportedUsageFlags & VK_IMAGE_USAGE_STORAGE_BIT)
         m_info.supportedUsage |= TextureUsage::UnorderedAccess;
-    }
+    if (surfaceCaps.supportedUsageFlags & VK_IMAGE_USAGE_SAMPLED_BIT)
```

Cppcheck (2.21.0)

[style] The function 'calcAlignedDown' is never used.

(unusedFunction)